docs/mathematics/factorial.mdx

```diff
@@ -0,0 +1,165 @@
+Factorial(n) = n * Factorial(n - 1)
+```
+
+Let us have a few examples:
+
+```
+Factorial(5) = 5 * 4 * 3 * 2 * 1 = 120
+Factorial(6) = 6 * 5 * 4 * 3 * 2 * 1 = 720 = 6 * Factorial(5)
+```
+
+## Algorithm
+
+### Recursive
+
+For a given number `n`,
+
+```
+factorial(n) = n * factorial(n - 1)
+```
+
+### Iterative
+
+For a given number `n`,
+
+Multiply numbers starting from `1` to `n` using a for loop.
+
+Return the factorial
+
+```
+factorial = 1
+for i = 1 to n
+  factorial = factorial * i
+return factorial
+```
+
+## Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+
+<TabItem value="cpp">
+
+```cpp
+//The length of vector should be >= length of the resultant factorial
+#include<bits/stdc++.h>
+using namespace std;
+
+/*The function takes a number and muiltiplies the number in the school's old fashioned
+way where we muiltiply the complete digit by one digit and then add the results to get
+the final resultant. The carry is carried forwarded and in the end the final result gets
+stored in the array/vector. */
+
+void factorial(int val)
+{
+    int carry = 0;       
+    vector <int> arr(10000, 0);   
+    arr[0] = 1;                   
+    int k = 0;                         
+
+    for(int i = 1; i <= val; i++) {
+        for(int j = 0;j <= k; j++) {      
+            arr[j] = arr[j] * i + carry;
+            carry = arr[j] / 10;
+            arr[j] = arr[j] % 10;
+        }
+        while(carry) {        
+            k++;
+            arr[k] = carry % 10;
+            carry /= 10;
+        }   
+    }
+    for(int i = k; i >= 0; i--) {  
+        cout<<arr[i];
+    }
+    cout << "\n";
+}
+int main() {
+    int num;
+    cout << "Enter the number: ";
+    cin>>num;
+    cout << "\nThe factorial of " << num << " is: ";
+    factorial(num);
+    return 0;
+}
+```
+
+</TabItem>
+<TabItem value="py">
+
+```python
+def fact(x):
+    result = 1
+    for i in range(1,x):
+        result = result * (i+1)
+    return result
+
+# Calculate factorial using recursion
+def fact_rec(x):
+    if x == 0 or x == 1:
+        return 1
+    else:
+        return x * fact_rec(x-1)
+
+
+if __name__ == "__main__":
+    n = int(input("Enter the number: "))
+    print("The factorial of {} is: {} ".format(n, fact(n)))
+```
+
+</TabItem>
+</Tabs>
+
+## Sample Input and Output
+
+### Input
+
+```
+Enter the number: 6
+```
+
+### Output
+
+```
+The factorial of 6 is: 720
+```
+
+## Complexity Analysis
+
+```
+Time complexity - O(n)
+Space complexity - O(1)
+```
+
+## Credits
+
+   - [Harsh Ghodkar](https://github.com/Saviour1001) for the [C++ implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/cp/long_factorials.cpp)
+   - [Ricardo Prins](https://github.com/ricardoprins) for the [Python implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/math/factorial.py)
```

Comment:
This was worked on by Rutuja: https://github.com/TesseractCoding/NeoAlgo/pull/952

Reply:
No no that's different. This one which I used was implemented by Ricardo.